ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -717,7 +728,9 @@ public void setAutoCapitalize(ReactEditText view, Dynamic autoCapitalize) {
   @ReactProp(name = "keyboardType")
   public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
     int flagsToSet = InputType.TYPE_CLASS_TEXT;
-    if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
+    if (keyboardType == null || !KEYBOARD_TYPES.contains(keyboardType)) {
```

Comment:
Why do we need this condition? Does it mean that `InputType.TYPE_CLASS_TEXT` will never be set? Can we just remove this condition and add a single else at the end?

Reply:
The `else` makes more sense. removes the need for the List check.

By default type is `InputType.TYPE_CLASS_TEXT`. Setting the flag again breaks the autoCapitalize flags.

Reply:
I cleaned up the logic and added a comment of why it is needed.

@makovkastar